File: apps/api/src/reports/dto/update-status.dto.ts
import { IsString, IsEnum, IsNotEmpty, IsOptional } from 'class-validator'
import { ReportStatus } from '@prisma/client'

export class UpdateStatusDto {
  @IsEnum(ReportStatus)
  @IsNotEmpty()
  status: ReportStatus

  @IsString()
  @IsNotEmpty()
  updatedBy: string

  @IsString()
  @IsOptional()
  comment?: string
}

File: apps/api-java/src/main/java/com/tickets/api/dto/report/UpdateStatusDto.java
package com.tickets.api.dto.report;

import com.tickets.api.model.enums.ReportStatus;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class UpdateStatusDto {
    @NotNull
    private ReportStatus status;
    private String comment;
    private String updatedBy; // user id
}
